client/pages/topsecret.jsx

```diff
@@ -1,43 +1,173 @@
 import React from 'react';
 import * as ReactBS from 'react-bootstrap';
-import {Navigation, Router} from 'react-router';
+import {Navigation} from 'react-router';
+import _ from 'lodash';
 
 const HomePage = React.createClass({
+
   mixins : [Navigation],
+
+  getInitialState() {
+    return {
+      initialBankSavings : 4294967296.69, initialExpenses : 0, tools : [],
+      addlbl : '', skuin : '', typein : '',
+      namein : '', costin : '', costQ : '', nameQ : '', searchdebug : '',
+      costfilter : '>',
+    };
+  },
+
   componentDidMount() {
     var {userid} = this.context.router.getCurrentQuery();
-    if(userid != 1)
-    {
-      //router.transitionTo('/');
+    if (userid != 1) {
+      // router.transitionTo('/');
       this.replaceWith('/');
-      console.log(this.context);
     }
 
+
   },
+  handleChange(key, event) {
+      this.setState({[key] : event.currentTarget.value});
+    },
+
+  handleAddClick(event) {
+      const doesExist = _.find(this.state.tools, { sku : this.state.skuin });
+      if (this.state.skuin == '' || this.state.namein == ''
```

Comment:
Triple equals

Another way to write this is:

```
if (! (this.state.skuin && this.state.namein && this.state.typein && this.state.costin)) {
   ...
}
```

If a string has no length it's considered "falsy"